File: x2/bernoulli.php
<?php
    // RUN FILE: 
    //          php bernoulli.php

    // Function that creates bernoulli number like the pseudocode
    function B($n): float {
        $B = array();   // initialize emtpy array
        $B[0] = 1;      // set first elem to 1

        // For-loop nr 1
        for ($m = 1; $m <= $n; $m++) {
            $B[$m] = 0;
            // For-loop nr 2
            for ($k = 0; $k <= $m-1; $k++) {
                $B[$m] = $B[$m] - binom($m+1, $k) * $B[$k];
            }
            $B[$m] = $B[$m]/($m+1);
        }        
        return $B[$n];
    }

    function binom($n, $k) {
        $r = 1;
        for ($i = 1; $i <= $k; $i++) {
            $r = $r * ($n - $i + 1)/$i;
        }
        return $r;
    }
    // Test binom: should output 10
    // echo binom(5, 2);
    // echo "\n";

    // $res = B(4); // output = -0.033333... = -1/30
    // echo "B(1) = ";
    // echo $res;
    // echo "\n";

    // Loop to print the first 10 Bernoulli numbers
    for ($n = 0; $n < 10; $n++) {
        $res = B($n);
        echo "B($n) = $res\n";
    }
?>

File: x2/bernoulli.clj
;; RUN FILE:
;;           clojure bernoulli.clj

; Func for binom which takes parameters n and k
(defn binom [n k]
  (let [r (atom 1.0)] ; define r = 1 as atom (muteable)
    ; loop in range 1 to k inclusive
    (doseq [i (range 1 (inc k))]
      ; swap changes atom-value with func value
      (swap! r (fn [current] (* current (/ (+ (- n i) 1) i)))))  ; current = r, divide () with i
    @r) ; @r dereferences variable r and returns r
)

; func that takes in n and returns the bernoulli number
(defn B [n]
  (let [B (atom (vec (repeat (inc n) 0.0)))] ; muteable vector B
    (swap! B assoc 0 1.0) ; set first element to 1
    ; loop m from 1 to n
    (doseq [m (range 1 (inc n))]
      (swap! B assoc m 0.0) ; B[m] = 0
      ; loop k from 0 to m-1 (not inc)
      (doseq [k (range 0 m)]
        (swap! B (fn [x] (assoc x m (- (get x m) (* (binom (inc m) k) (get x k)))))) ; calculate
        )
      (swap! B update m (fn [x] (/ (double x) (+ 1 m)))) ; divide with m+1
      )
    (get @B n)) ; retrieve output
  )

;; (println (binom 5 2))
;; (println (B 4)) ; should return -0.03333... = (-1/30)

(doseq [n (range 10)]
  (println (str "B(" n ") = " (B n))))